views/forms/TextInput/index.jsx

```diff
@@ -1,30 +1,24 @@
 import React from 'react'
 import TextField from '@mui/material/TextField'
-import { Controller, useFormContext } from 'react-hook-form'
+import { useController } from 'react-hook-form'
 
 const TextInput = (props) => {
-  const { name, onChange, ...rest } = props
-  const { control } = useFormContext()
+  const { errors } = props
+  const { field } = useController(props)
 
   return (
-    <Controller
-      name={name}
-      control={control}
-      render={({ field }) => (
-        <TextField
-          aria-label={name}
-          fullWidth={true}
-          margin="dense"
-          {...field}
-          {...rest}
-          inputRef={field.ref}
-          onChange={(e) => {
-            field.onChange(e)
-
-            if (onChange) onChange(e)
-          }}
-        />
-      )}
+    <TextField
+      aria-invalid={!!errors ? 'true' : 'false'}
+      aria-label={props.label}
+      fullWidth={props.fullWidth}
+      helperText={errors?.message}
+      InputLabelProps={{ shrink: true }}
+      inputProps={props.inputProps}
+      label={props.label}
+      margin="normal"
+      required={props.required}
+      type={props.type}
+      {...field}
```

Comment:
Is it necessary to have all of these different props. What concerns me is if we want to change something like the `margin` from `normal` to `dense` then we don't have the capacity to do that, or having to set the fullWidth property all the time. If we de-structure the necessary props and then spread the rest we will probably have less configurations on this one component.

Reply:
I usually prefer being explicit about props rather than spreading them because when you spread props you could unknowingly pass props that are not used, or even worse are used to an unintended effect, by the child component. To your point, `fullWidth` does not need to be passed, it is an optional prop that can be undefined. I think we could have a sane default for `margin`, though, in case we want something other than `normal`.